lightning/src/ln/wire.rs

```diff
@@ -22,11 +22,30 @@ use io;
 use ln::msgs;
 use util::ser::{Readable, Writeable, Writer};
 
+/// Trait to be implemented by custom message (unrelated to the channel/gossip LN layers)
+/// decoders.
+pub trait CustomMessageReader {
+	/// The type of the message decoded by the implementation.
+	type CustomMessageType;
+	/// Decodes a custom message to `CustomMessageType`. If the given message type is known to the implementation and
+	/// the message could be decoded, must return `Ok(Some(message))`. If the message type
+	/// is unknown to the implementation, must return `Ok(None)`. If a decoding error
+	/// occur, must return `Err(DecodeError::X)` where `X` details the encountered error.
+	fn read<R: io::Read>(&self, message_type: u16, buffer: &mut R) -> Result<Option<Self::CustomMessageType>, msgs::DecodeError>;
+}
+
+/// Handler for messages external to the LN protocol.
+pub trait CustomMessageHandler : CustomMessageReader {
```

Comment:
This doesn't need to be a subtrait since `peer_handler` uses both as a trait bound.

Reply:
See https://github.com/rust-bitcoin/rust-lightning/pull/1031#discussion_r685661669